model-engine/model_engine_server/inference/batch_inference/vllm_batch.py

```diff
@@ -0,0 +1,209 @@
+import asyncio
+import json
+import os
+import subprocess
+import time
+from urllib.parse import urlparse
+
+import boto3
+import smart_open
+from model_engine_server.common.dtos.llms import (
+    CompletionOutput,
+    CreateBatchCompletionsRequest,
+    CreateBatchCompletionsRequestContent,
+    TokenOutput,
+)
+
+CONFIG_FILE = os.getenv("CONFIG_FILE")
+AWS_REGION = os.getenv("AWS_REGION", "us-west-2")
+
+os.environ["AWS_PROFILE"] = os.getenv("S3_WRITE_AWS_PROFILE", "default")
+
+
+def get_s3_client():
+    session = boto3.Session(profile_name=os.getenv("S3_WRITE_AWS_PROFILE"))
+    return session.client("s3", region_name=AWS_REGION)
+
+
+def download_model(checkpoint_path, final_weights_folder):
+    s5cmd = f"./s5cmd --numworkers 512 sync --concurrency 10 {os.path.join(checkpoint_path, '*')} {final_weights_folder}"
+    process = subprocess.Popen(
+        s5cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+    )
+    for line in process.stdout:
+        print(line)
+
+    process.wait()
+
+    if process.returncode != 0:
+        stderr_lines = []
+        for line in iter(process.stderr.readline, ""):
+            stderr_lines.append(line.strip())
+
+        raise IOError(f"Error downloading model weights: {stderr_lines}")
```

Comment:
curious: wondering why you wouldn't run this with check=True for the subprocess? Seems like it'd be less verbose. But I'm guessing I'm missing something, just want to understand Popen a bit better.

Reply:
`subprocess.Popen` doesn't have this option. i use Popen to stream output